src/main/java/com/skynest/uitesting/pages/BasePage.java

```diff
@@ -0,0 +1,21 @@
+package com.skynest.uitesting.pages;
+
+import lombok.AllArgsConstructor;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.support.ui.WebDriverWait;
+
+import static com.skynest.uitesting.constants.PageUrlConstants.LOGIN_URL;
+
+@AllArgsConstructor
+public class BasePage {
+    protected final WebDriver driver;
+    protected final WebDriverWait wait;
+
+    public void openPage() {
```

Comment:
Leaving just the method that accepts the url argument is perfectly fine. Otherwise, you will be able to call the openPage() method on the BasePage method and, by extension, on all future subclasses of BasePage. But why would, say "filePage.openPage()" lead you to the login page? It's not intuitive,

Reply:
Implemented the second option.